src/Type/Accessory/AccessoryArrayListType.php

```diff
@@ -239,7 +239,7 @@ public function getLastIterableValueType(): Type
 
 	public function isArray(): TrinaryLogic
 	{
-		return TrinaryLogic::createYes();
+		return TrinaryLogic::createMaybe();
```

Comment:
I haven't checked the rest of this PR but this is certainly wrong. If something is a list then we know it's an array...

Reply:
I understand you're changing this so that some other algorithm results in an outcome we want, but the typesystem layer always has to tell the truth, otherwise something else breaks...

Reply:
I'm likely to accept whatever you come up with, but make sure to do some benchmarks too :) It's really easy to think you've sped things up, but in reality you might have slowed them down :) Thanks.